valet-key/ValetKey.Web/Controllers/HomeController.cs

```diff
@@ -50,38 +55,34 @@ public ActionResult Index()
             return this.View();
         }
 
+        /// <summary>
+        /// We return a limited access key that allows the caller to download a file to this specific destination for defined period of time
+        /// </summary>
         private async Task<StorageEntitySas> GetSharedAccessReferenceForDownload(string blobName)
         {
-            var blobClient = this.account.CreateCloudBlobClient();
-            var container = blobClient.GetContainerReference(this.blobContainer);
-
-            var blob = container.GetBlockBlobReference(blobName);
-            
-            if (!await blob.ExistsAsync())
-            {
-                throw new Exception("Blob does not exist");
-            }
+            var blob = blobServiceClient.GetBlobContainerClient(this.blobContainer).GetBlobClient(blobName);
 
-            var policy = new SharedAccessBlobPolicy
-            {
-                Permissions = SharedAccessBlobPermissions.Read,
+            var storageSharedKeyCredential = new StorageSharedKeyCredential(blobServiceClient.AccountName, ConfigurationManager.AppSettings["AzureStorageEmulatorAccountKey"]);
 
-                // Create a signature for 5 min earlier to leave room for clock skew
-                SharedAccessStartTime = DateTime.UtcNow.AddMinutes(-5),
+            var blobSasBuilder = new BlobSasBuilder
 
-                // Create the signature for as long as necessary -  we can 
-                SharedAccessExpiryTime = DateTime.UtcNow.AddMinutes(5)
+            {
+                BlobContainerName = this.blobContainer,
+                BlobName = blobName,
+                Resource = "b",
+                StartsOn = DateTimeOffset.UtcNow,
```

Comment:
As with the later code, we've made some subtle changes from the previous code with respect to the validity time range for the signature.  Was this intentional?